### PR TITLE
Update Core-setup to 2.0.0-preview2-25303-01

### PR DESCRIPTION
**Pull request by chcosta**
This is testing out the core-setup on buildtools changes in the cli.

### CODE REVIEW

**Comment by chcosta** [comment]
I'll need to investigate how the cli determines dependency names.  We removed the `-portable` moniker from asset names in core-setup

```
Task "DownloadFile" (TaskId:17)
17:40:15                      Task Parameter:Uri=https://dotnetcli.azureedge.net/dotnet/master/Binaries/2.0.0-preview2-25303-01/dotnet-linux-x64.2.0.0-preview2-25303-01-portable.tar.gz
```

**Comment by chcosta** [comment]
Now we're failing due to the known issue of missing `.deb` packages.  I'll leave this PR open for the moment while we are bringing up the .deb packages

**Comment by eerhardt** [comment]
> I'll need to investigate how the cli determines dependency names. We removed the -portable moniker from asset names in core-setup

I can fix it.

**Comment by eerhardt** [comment]
@chcosta - I fixed the test that was failing on Windows.

File: build/BundledRuntimes.props [hunk 1]
```diff
@@ -1,23 +1,20 @@
 <Project>
   <PropertyGroup>
-    <ArchivePortableSuffix Condition=" '$(UsePortableSharedFramework)' == 'true' ">-portable</ArchivePortableSuffix>
-    <InstallerPortableSuffix Condition=" '$(UsePortableSharedFramework)' == 'true' AND '$(OSName)' != 'ubuntu' ">-portable</InstallerPortableSuffix>
-
     <!-- Downloaded Installers + Archives -->
-    <DownloadedSharedHostInstallerFileName Condition=" '$(InstallerExtension)' != '' ">dotnet-host-$(ProductMonikerRid).$(SharedHostVersion)$(InstallerPortableSuffix)$(InstallerExtension)</DownloadedSharedHostInstallerFileName>
+    <DownloadedSharedHostInstallerFileName Condition=" '$(InstallerExtension)' != '' ">dotnet-host-$(ProductMonikerRid)-$(SharedHostVersion)$(InstallerExtension)</DownloadedSharedHostInstallerFileName>
```
Comment (eerhardt) [inline build/BundledRuntimes.props]: I've just noticed that we have changed format of the asset name to use a `-` instead of a `.` separator between the RID and version.

The CLI still uses `.`. We need to be consistent across the SDK and the runtime assets on usage of `.` and `-` separators and format of the assets.  This is going to break the `dotnet-install` script as well.  See [https://github.com/dotnet/cli/blob/master/scripts/obtain/dotnet-install.ps1#L232](https://github.com/dotnet/cli/blob/master/scripts/obtain/dotnet-install.ps1#L232)

@chcosta - why was this change made to the assets?  I think we need to revert it in core-setup.  There will be too many things that will break, and I don't feel this is worth the breaking change.

/cc @livarcocc

**Comment by chcosta** [comment]
I think this was an unintentional change which got propagated around. I'll look at fixing this

**Comment by chcosta** [comment]
Yes, I've been looking into the repercussions of this a bit, and I agree. this minor change could cause lots of headaches.  I'll need to clean the azure storage of cruft again, but that shouldn't be a terrible price to move this back to a tested state.

**Comment by chcosta** [comment]
This PR lost the fix for the windows break.  Also, packages weren't published to myget yet. so we're not quite ready for this.

**Comment by eerhardt** [comment]
> This PR lost the fix for the windows break.

fixed.

**Comment by chcosta** [comment]
```
17:12:23 [39;49m[39;49m[30m                     Downloading 'https://dotnetcli.azureedge.net/dotnet/master/Installers/2.0.0-preview2-25305-19/dotnet-sharedframework-ubuntu-x64.2.0.0-preview2-25305-19.deb' to '/opt/code/artifacts/ubuntu.14.04-x64/packages/dotnet-sharedframework-ubuntu-x64.2.0.0-preview2-25305-19.deb' (TaskId:58)
```

**Comment by chcosta** [comment]
@eerhardt , is ubuntu.14.04 supposed to produce zips as `ubuntu`, not `ubutnu..14.04`?

**Comment by eerhardt** [comment]
Yes. Just like the .version files, Ubuntu 14 and Debian moniker names aren't the same as the RID, unfortunately.

In this case, these are just the installer names, which the dotnet-install scripts don't use. So maybe we can fix this one by changing the CLI to expect the new names.

**Comment by eerhardt** [comment]
I see the issue. The dependencies of DependencyModel NuGet package are messed up. Instead of depending on 2.0.0-preview2-12345 of PlatformAbstractions, the nupkg says it depends on "2.0.0" of PlatformAbstractions, which doesn't exist yet.

I can fix this early Monday morning.

**Comment by eerhardt** [comment]
@chcosta, if you want to revert the versions of DependencyModel and PlatformAbstractions in this PR, then we can test the rest of the changes. 

Also- are you fixing the Ubuntu 14 and Debian download file issue?

**Comment by chcosta** [comment]
Ok, i'll revert that part of the change.

I should have time to look into the Ubuntu / Debian issue a bit later today.

**Comment by eerhardt** [comment]
I guess it is just Ubuntu 14.

**Comment by chcosta** [comment]
I'm surprised Debian passed.  I expected Debian.8 -> Debian.

**Comment by eerhardt** [comment]
I had forgotten that the only Linux distro we have Deb packages for in the CLI is Ubuntu 14. We don't have 16.04, 16.10 or Debian

**Comment by eerhardt** [comment]
@chcosta - can you take a look at my latest commit?

**Comment by eerhardt** [comment]
I've seen the Ubuntu failures before.  I haven't gotten a chance to look at it (will tonight or early tomorrow morning).  We may (probably) need a fix in core-setup for this.

**Comment by eerhardt** [comment]
The ubuntu failure is because our `build\netcoreapp2.0\Microsoft.NETCore.App.PlatformManifest.txt` file isn't correct on the latest master builds of core-setup.

We aren't pulling in all the native assets across platforms, just the windows ones.

Notice there are no `.so` or `.dylib` files in:

[Microsoft.NETCore.App.PlatformManifest.txt](https://github.com/dotnet/cli/files/985366/Microsoft.NETCore.App.PlatformManifest.txt)

I'll get a fix for this in core-setup.

/cc @ericstj

**Comment by chcosta** [comment]
Woohoo!

**Comment by eerhardt** [comment]
@livarcocc - can you get someone from the CLI side to review?  And then we can put this in and get the ball rolling on regular updates in master again.

File: build/BundledRuntimes.props [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
+    <DownloadedSharedFrameworkInstallerFileName Condition=" '$(InstallerExtension)' != '' ">dotnet-sharedframework-$(CoreSetupRid).$(SharedFrameworkVersion)$(InstallerExtension)</DownloadedSharedFrameworkInstallerFileName>
     <DownloadedSharedFrameworkInstallerFile Condition=" '$(InstallerExtension)' != '' ">$(PackagesDirectory)/$(DownloadedSharedFrameworkInstallerFileName)</DownloadedSharedFrameworkInstallerFile>
 
     <!-- Use the portable linux-x64 Rid when downloading the shared framework compressed file.
          NOTE: There isn't a 'linux-x64' version of the installers or the additional shared framweork. -->
-    <SharedFrameworkRid>$(ProductMonikerRid)</SharedFrameworkRid>
+    <SharedFrameworkRid>$(CoreSetupRid)</SharedFrameworkRid>
     <SharedFrameworkRid Condition=" '$(UsePortableLinuxSharedFramework)' == 'true' ">linux-x64</SharedFrameworkRid>
     <CombinedFrameworkHostCompressedFileName>dotnet-$(SharedFrameworkRid).$(SharedFrameworkVersion)$(ArchivePortableSuffix)$(ArchiveExtension)</CombinedFrameworkHostCompressedFileName>
```
Comment (eerhardt) [inline build/BundledRuntimes.props]: @chcosta - looks like here is one place we missed.  There is still a reference to `$(ArchivePortableSuffix)`.  I'll fix this up.

File: test/dotnet-new.Tests/GivenThatIWantANewApp.cs [hunk 3]
```diff
@@ -118,7 +118,7 @@ string GetFrameworkPackageVersion()
                     var context = reader.Read(stream);
                     var dependency = context
                         .RuntimeLibraries
-                        .Single(library => library.Name == packageName);
+                        .Single(library => string.Equals(library.Name, packageName, StringComparison.OrdinalIgnoreCase));
```
Comment (livarcocc) [inline test/dotnet-new.Tests/GivenThatIWantANewApp.cs]: Why is this needed?

Comment (eerhardt) [inline test/dotnet-new.Tests/GivenThatIWantANewApp.cs]: Because the sharedfx's .deps.json file is now being written using a new DependencyModel that doesn't have the "toLower" hack in it.  Thus the Microsoft.NETCore.App.deps.json file now has the normal casing in it. Here the test was expecting the `netstandard.library` library name to be all lower-case, which is not a good assumption.  These are case-insensitive.

**Comment by eerhardt** [comment]
Test CentOS7.1 x64 Debug Build
